src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/PyrisJobService.java

```diff
@@ -113,6 +116,19 @@ public void removeJob(String token) {
         jobMap.remove(token);
     }
 
+    /**
+     * Checks if a job with the specified course, lecture, and lecture unit IDs exists in the job map.
+     *
+     * @param courseId      The ID of the course to check.
+     * @param lectureId     The ID of the lecture to check.
+     * @param lectureUnitId The ID of the lecture unit to check.
+     * @return true if a matching IngestionWebhookJob exists; false otherwise.
+     */
+    public boolean jobExists(long courseId, long lectureId, long lectureUnitId) {
```

Comment:
This method has a very general name but a very specific purpose. I would prefer to see a method like `currentJobs()` which returns `jobMap.values()` and then you do your required filtering at the call-site. This keeps the code in `PyrisJobService` general-purpose.